### PR TITLE
Changing CallOut mediator to support distributed transaction.

### PR DESCRIPTION
**Pull request by VijithaEkanayake**

### CODE REVIEW

File: modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java [hunk 1]
```diff
@@ -365,6 +365,42 @@ public static void beginTransaction() throws Exception {
...
+		long key = Thread.currentThread().getId();
+		try {
+			if (log.isDebugEnabled()) {
+				log.debug("beginTransaction()");
+			}
+
+			TransactionManager txMgr = txManagers.get().get(key);
+			return txMgr;
+		} catch (Exception ex) {
+			log.debug(" BEGIN ERROR  : " + txManagers.get().get(key).getStatus());
```
Comment (Buddhima) [inline modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java]: this is an error log ( log.error(...) )

File: modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java [hunk 2]
```diff
@@ -365,6 +365,42 @@ public static void beginTransaction() throws Exception {
...
+			}
+
+			TransactionManager txMgr = txManagers.get().get(key);
+			txMgr.begin();
+			Transaction tx = txMgr.getTransaction();
+			transactions.get().put(key, tx);
+			return  tx;
+
+		} catch (Exception ex) {
+			log.debug(" BEGIN ERROR  : " + txManagers.get().get(key).getStatus());
```
Comment (Buddhima) [inline modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java]: same here, log.error(...)

File: modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java [hunk 3]
```diff
@@ -365,6 +365,42 @@ public static void beginTransaction() throws Exception {
 		}
 
 	}
+
+	public static TransactionManager getTransactionManager() throws Exception {
+		long key = Thread.currentThread().getId();
+		try {
+			if (log.isDebugEnabled()) {
+				log.debug("beginTransaction()");
```
Comment (virajsenevirathne) [inline modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java]: formatting issues

File: modules/core/src/main/java/org/apache/synapse/mediators/builtin/CalloutMediator.java [hunk 4]
```diff
@@ -153,6 +160,25 @@ public boolean mediate(MessageContext synCtx) {
                 }
             }
 
+            if(this.serviceURL.contains("transport.jms.TransactionCommand=begin")) {
```
Comment (virajsenevirathne) [inline modules/core/src/main/java/org/apache/synapse/mediators/builtin/CalloutMediator.java]: Please put this as a static constant

**Pull request merged**